provider/pkg/gen/python-templates/yaml/yaml.tmpl

```diff
@@ -410,7 +402,26 @@ def _read_file(path: str) -> str:
 def _build_resources_dict(objs: Sequence[pulumi.Output]) -> Mapping[pulumi.Output, pulumi.Output]:
     return {key: value for key, value in objs}
 
-
+def _get_child_options(parent: pulumi.Resource, opts: pulumi.ResourceOptions):
```

Comment:
It would be nice to add type hints on the return types for these new functions as well.